server/src/main/java/org/elasticsearch/action/admin/indices/rollover/TransportRolloverAction.java

```diff
@@ -221,6 +232,35 @@ protected void masterOperation(
 
             listener.delegateFailureAndWrap((delegate, statsResponse) -> {
 
+                AutoShardingResult rolloverAutoSharding = null;
+                final IndexAbstraction indexAbstraction = clusterState.metadata()
+                    .getIndicesLookup()
+                    .get(rolloverRequest.getRolloverTarget());
+                if (indexAbstraction.getType().equals(IndexAbstraction.Type.DATA_STREAM)) {
+                    DataStream dataStream = (DataStream) indexAbstraction;
+                    final Optional<IndexStats> indexStats = Optional.ofNullable(statsResponse)
```

Comment:
Minor nit, but I don't think `statsResponse` can be null, so this could be
```suggestion
                    final Optional<IndexStats> indexStats = Optional.of(statsResponse)
```